### PR TITLE
feat(Combobox): add clearValue prop for custom reset value on clear

### LINKED ISSUE #2565: [Feature]: Allow specifying clear value when `resetModelValueOnClear` is set in `<Combobox>`
### Describe the feature

Currently, when combobox gets cleared with the `resetModelValueOnClear` set to true and no `multiple` prop, model value is always set to `null`. It would be great to add a prop, e.g. `clearValue` (just as an idea), which would be used in this scenario when specified by the user.

Context:
https://github.com/unovue/reka-ui/issues/1654#issuecomment-4134464202
https://github.com/nuxt/ui/pull/5643#issuecomment-3796828151

### Additional information

- [ ] I intend to submit a PR for this feature.
- [ ] I have already implemented and/or tested this feature.

### PR DESCRIPTION
**Pull request by yan-ad**
### 🔗 Linked issue

Resolves #2565

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Adds a `clearValue` prop to `ComboboxRoot` that allows specifying a custom value to use when resetting the model value on clear (via `ComboboxCancel` with `resetModelValueOnClear: true`).

**Before:** clearing always reset to `null` (single) or `[]` (multiple).
**After:** when `clearValue` is set, single-select clears to that value instead of `null`. Multiple mode always resets to `[]` regardless of `clearValue` (by design — an empty array is the natural "nothing selected" state for multi-select).

**Changes:**
- `ComboboxRoot.vue` — add `clearValue` prop to `ComboboxRootProps` interface and `ComboboxRootContext` type
- `ComboboxCancel.vue` — use `clearValue` from context when resetting modelValue (falls back to `null` when not set)
- `AutocompleteRoot.vue` — add `clearValue: ref(undefined)` to provided Combobox context
- `Combobox.test.ts` — add 3 tests: custom clearValue, default null fallback, multiple mode ignores clearValue

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

Tests: 45/45 pass (`npx vitest run src/Combobox/Combobox.test.ts`)

## Summary by CodeRabbit

* **New Features**
  * Added a `clearValue` prop to Combobox components to customize what `modelValue` is set to when cleared.

* **Bug Fixes**
  * Refined clear/cancel behavior so clearing resets `modelValue` to `clearValue` in single-select mode (or `null` when unset), and to `[]` in multiple-select mode.

* **Tests**
  * Added coverage for clear/cancel scenarios, including behavior with and without `clearValue`, and in multiple-select mode.

* **Documentation**
  * Updated `ComboboxRoot` prop docs and added an example demonstrating custom clear values.

### CODE REVIEW

File: packages/core/src/Combobox/ComboboxRoot.vue [hunk 1]
```diff
@@ -77,6 +78,12 @@ export interface ComboboxRootProps<T = AcceptableValue> extends Omit<ListboxRoot
    * When `true` the `modelValue` will be reset to `null` (or `[]` if `multiple`)
    */
   resetModelValueOnClear?: boolean
+  /**
+   * The value to set the `modelValue` to when the combobox is cleared
+   * and `resetModelValueOnClear` is `true`. When not provided, defaults to
+   * `null` for single select or `[]` for multiple select.
+   */
+  clearValue?: T
 }
```
[the remark this answers is not in the document]
Comment (yan-ad) [reply packages/core/src/Combobox/ComboboxRoot.vue:87]: Regenerated API docs with `pnpm docs:gen` in 449b27b — `clearValue` prop is now properly documented in `ComboboxRoot.md`.

**Review by yan-ad — COMMENTED** [review]
Addressed the nitpick — clarified in docs that `clearValue` only applies to single-select mode, not multiple.

**Review by zernonia — CHANGES_REQUESTED** [review]
Thanks @yan-ad, the implementation and tests are tidy!

One concern: the request behind #2565 (see the linked #1654 and nuxt/ui threads) is to clear to `undefined` instead of `null`. Because the check is `clearValue !== undefined`, and Vue can't tell an explicit `undefined` from an omitted prop, `:clear-value="undefined"` still resets to `null`. Could we support that case, e.g. by checking whether the prop was actually passed, or by accepting a factory? If not, it may be simpler to handle it consumer-side.

Also, the docs example resets to `"No one selected"`, which isn't one of the options and ends up shown in the input as if it were a selection. Could we use a more realistic default value there?
